fix(subheader/sticky): Stick subheader to content instead of parent.

topherfangio opened the pull request:
The `$mdSticky` service used to stick it's contents to the parent of the `<md-content>` container. This resulted in some odd styling issues and the inability to have two lists with sticky subheaders side-by-side.

This fix appends the subheaders directly to the `<md-content>` to fix this issue.

Also, updated subheaders to use `<div>` tags instead of `<h2>`.

fixes #3535

topherfangio commented:
@jelbourn @ThomasBurleson @rschmukler  Can you guys please review this and let me know if you see any issues with the updates? This is related to the discussion we had in Monday's meeting (#3535), but I found a way to do it without the visual glitch I mentioned.

jelbourn commented on src/components/sticky/sticky.js:
```diff
@@ -169,22 +171,22 @@ function MdSticky($document, $mdConstant, $compile, $$rAF, $mdUtil) {
-      } else if (isScrollingDown && self.next) {
-        if (self.next.top - scrollTop <= 0) {
+      } else if (isScrollingDown) {
+        if (self.next && self.next.top - scrollTop <= 0) {
```
I know it was like this before, but could you change this? 
`self.next.top - scrollTop <= 0` is a less readable way of saying `self.next.top <= scrollTop`

jelbourn commented on src/components/sticky/sticky.js:
```diff
@@ -169,22 +171,22 @@ function MdSticky($document, $mdConstant, $compile, $$rAF, $mdUtil) {
       // Scrolling up with a current sticky item?
-      } else if (!isScrollingDown && self.current) {
-        if (scrollTop < self.current.top) {
+      } else if (!isScrollingDown) {
```
This can just be an `else` now.

jelbourn commented on src/components/sticky/sticky.js:
```diff
@@ -169,22 +171,22 @@ function MdSticky($document, $mdConstant, $compile, $$rAF, $mdUtil) {
       // Going to next item?
-      } else if (isScrollingDown && self.next) {
-        if (self.next.top - scrollTop <= 0) {
+      } else if (isScrollingDown) {
```
It feels like this whole block could be a better high-level comment explaining what's going on.

topherfangio commented:
@jelbourn @ThomasBurleson As requested, I have refactored the `onScroll()` function and it's comments quite a bit to increase readability/understanding. In particular, I replaced the `if/else if` blocks with more simple `if/return` statements to more clearly separate the different cases and remove some duplicate code.

Can you two please take a look and provide feedback? It may be easier if you look at the [whole file](https://github.com/angular/material/blob/fix-subheader-sticky-3535/src/components/sticky/sticky.js#L164) rather than the diff so you can see the overall structure.
